src/react/components/pages/prebuiltPredict/prebuiltPredictResult.tsx

```diff
@@ -0,0 +1,198 @@
+        for (const name of Object.keys(predictions)) {
+            const prediction = predictions[name];
+            if (prediction != null) {
+                prediction.fieldName = name;
+                prediction.displayOrder = tagsDisplayOrder.indexOf(name);
+            }
+        }
+        // not sure if we decide to filter item by the page
+        const items = Object.values(predictions).filter(Boolean).sort((p1, p2) => p1.displayOrder - p2.displayOrder);
+
+        return (
+            <div>
+                <div className="container-items-center container-space-between results-container">
+                    <h5 className="results-header">Prediction results</h5>
+                    <PrimaryButton
+                        className="align-self-end keep-button-80px"
+                        theme={getPrimaryGreenTheme()}
+                        text="Download"
+                        allowDisabledFocus
+                        autoFocus={true}
+                        onClick={this.onDownloadClick}
+                    />
+                </div>
+                <div className="prediction-field-header">
+                    <h6 className="prediction-field-header-field"> Page # / Field name / Value</h6>
+                    <h6 className="prediction-field-header-confidence"> Confidence</h6>
+                </div>
+                <div className="prediction-header-clear"></div>
+
+                {items.map((item: any, key) => this.renderItem(item, key))}
+            </div>
+        );
+    }
+
+    private renderItem = (item: any, key: any) => {
+        const postProcessedValue = getPostProcessedValue(item);
+        const style: any = {
+            marginLeft: "0px",
+            marginRight: "0px",
+            background: this.getTagColor(item.fieldName),
+        };
+        return (
+            <div key={key}
+                onClick={() => this.onPredictionClick(item)}
+                onMouseEnter={() => this.onPredictionMouseEnter(item)}
+                onMouseLeave={() => this.onPredictionMouseLeave(item)}>
+                <li className="predictiontag-item" style={style}>
+                    <div className={"predictiontag-color"}>
+                        <span>{item.page}</span>
+                    </div>
+                    <div className={"predictiontag-content"}>
+                        {this.getPredictionTagContent(item)}
+                    </div>
+                </li>
+                <li className={postProcessedValue ? "predictiontag-item-label mt-0" : "predictiontag-item-label mt-0 mb-1"}>
+                    {postProcessedValue ? "text: " + item.text : item.text}
+                </li>
+                {postProcessedValue &&
+                    <li className="predictiontag-item-label mb-1">
+                        {postProcessedValue}
+                    </li>
+                }
+            </div>
+        );
+    }
+
+    private getTagColor = (name: string): string => {
```

Comment:
An example of duplicate code that we want to refactor out. This is the same function in existing predictResult.tsx class